src/build_context.rs

```diff
@@ -24,7 +24,9 @@ pub enum BridgeModel {
     /// A native module with c bindings, i.e. `#[no_mangle] extern "C" <some item>`
     Cffi,
     /// A rust binary to be shipped a python package
-    Bin,
+    /// The String is the name of the bindings
+    /// providing crate, e.g. pyo3, the number is the minimum minor python version
+    Bin(Option<(String, usize)>),
```

Comment:
@davidhewitt I wonder should we support **abi3** bin bindings, if so I think `BridgeModel` needs a refactoring to separate `bin`/`lib` targets from `pyo3`/`rust-cpython` bindings crate. Something like

```rust
enum BuildType {
    Bin(Option<BridgeModel>),
    Library(BridgeModel),
}

enum BridgeModel {
    Cffi,
    Bindings(String, usize),
    BindingsAbi3(u8, u8),
}
```

Reply:
Based on https://github.com/PyO3/pyo3/pull/2328#issuecomment-1107871639 I understood that it's not really possible to link to `libpython3.so` (i.e. abi3 distribution) without fixing issues in upstream CPython.